### PR TITLE
Custom accessibleRole, accessibleName, accessibleDescription for controls in Accessibility test app

### PR DESCRIPTION
**Pull request by Amy-Li03**
In this PR, we custom accessibleRole, accessibleName, accessibleDescription for all controls which supported these properties.

## Proposed changes

- Create a new form in Accessibility test solution, add all supported controls to form and custom their accessibleRole, accessibleName, accessibleDescription properties.

## Customer Impact
No

## Regression? 

 No

## Risk
No

## Test methodology 

- Manual testing using Inspect and Accessibility Insight

## Test environment(s) 

- .Net SDK 6.0.100-alpha.1.20623.10

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/4432)

### CODE REVIEW

File: src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Windows.Forms;
+
+namespace AccessibilityTests
+{
+    public partial class CusAccNameTesting : Form
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.cs:9]: We don't have any constraints on the length of the type, so let's give it a clearer and more meaningful name, for example:
```suggestion
    public partial class CustomAccessiblePropertiesForm  : Form
```

File: src/System.Windows.Forms/tests/AccessibilityTests/Main.Designer.cs [hunk 2]
```diff
@@ -129,11 +130,22 @@ private void InitializeComponent()
             this.btnDataBindingExample.UseVisualStyleBackColor = true;
             this.btnDataBindingExample.Click += new System.EventHandler(this.btnDataBindingExample_Click);
             // 
+            // BtnCusAccName
+            // 
+            this.btnCusAccName.Location = new System.Drawing.Point(12, 410);
+            this.btnCusAccName.Name = "cusAccName";
+            this.btnCusAccName.Size = new System.Drawing.Size(455, 23);
+            this.btnCusAccName.TabIndex = 9;
+            this.btnCusAccName.Text = "Testing controls whose accessible name, role, description have been customized";
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/AccessibilityTests/Main.Designer.cs:139]: To make it a little more consistent with the rest:
```suggestion
            this.btnCusAccName.Text = "Accessible properties: Testing controls with custom accessible name, role, and description";
```

File: src/System.Windows.Forms/tests/AccessibilityTests/Main.Designer.cs [hunk 3]
```diff
@@ -129,11 +130,22 @@ private void InitializeComponent()
             this.btnDataBindingExample.UseVisualStyleBackColor = true;
             this.btnDataBindingExample.Click += new System.EventHandler(this.btnDataBindingExample_Click);
             // 
+            // BtnCusAccName
+            // 
+            this.btnCusAccName.Location = new System.Drawing.Point(12, 410);
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/AccessibilityTests/Main.Designer.cs:135]: This is a button for testing "custom accessible name, role, and description".
So I would prefer to rename this variable to **"customAccessiblePropertiesButton"** or something like that
/cc: @RussKie

File: src/System.Windows.Forms/tests/AccessibilityTests/Main.cs [hunk 4]
```diff
@@ -67,5 +67,12 @@ private void btnDataBindingExample_Click(object sender, EventArgs e)
             DataBindingExample dataBindingExample = new DataBindingExample();
             dataBindingExample.Show();
         }
+
+        private void btnCusAccName_Click(object sender, EventArgs e)
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/AccessibilityTests/Main.cs:71]: Please rename this handler too to have a clearer name

File: src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.cs (hunk 1 above)
Comment (vladimir-krestov) [reply src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.cs:9]: Don't forger rename files too 😊

File: src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs [hunk 5]
```diff
@@ -0,0 +1,1219 @@
+﻿
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs:1]: @RussKie Why don't we need to have the License strings here?

File: src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs [hunk 6]
```diff
@@ -0,0 +1,1219 @@
...
+            this.toolStripContainer1.Text = "toolStripContainer1";
+            // 
+            // 
+            // 
+            this.toolStripContainer1.TopToolStripPanel.Location = new System.Drawing.Point(0, 0);
+            this.toolStripContainer1.TopToolStripPanel.Name = "";
+            this.toolStripContainer1.TopToolStripPanel.Orientation = System.Windows.Forms.Orientation.Horizontal;
+            this.toolStripContainer1.TopToolStripPanel.RowMargin = new System.Windows.Forms.Padding(3, 0, 0, 0);
+            // 
+            // Form1
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs:1059]: You can set custom accessibility values for the Form too

File: src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs [hunk 7]
```diff
@@ -0,0 +1,1219 @@
...
+            // 
+            this.AutoScaleDimensions = new System.Drawing.SizeF(7F, 15F);
+            this.AutoScaleMode = System.Windows.Forms.AutoScaleMode.Font;
+            this.ClientSize = new System.Drawing.Size(914, 817);
+            this.ContextMenuStrip = this.contextMenuStrip1;
+            this.Controls.Add(this.dataGridView1);
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs:1065]: Please add `RichTextBox` too

File: src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs [hunk 8]
```diff
@@ -0,0 +1,1219 @@
...
+            this.Controls.Add(this.flowLayoutPanel1);
+            this.Controls.Add(this.domainUpDown1);
+            this.Controls.Add(this.dateTimePicker1);
+            this.Controls.Add(this.comboBox1);
+            this.Controls.Add(this.checkedListBox1);
+            this.Controls.Add(this.checkBox1);
+            this.Controls.Add(this.button1);
+            this.Margin = new System.Windows.Forms.Padding(4, 3, 4, 3);
+            this.Name = "CustomAccPropertyForm";
+            this.Text = "CustomAccPropertyForm";
```
Comment (vladimir-krestov) [inline src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs:1103]: Please rename them too

**Review by vladimir-krestov — APPROVED** [review]
Everything is good, except naming

File: src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs (hunk 7 above)
Comment (Amy-Li03) [reply src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs:1065]: Hi @vladimir-krestov, for RichTextBox control, there is a issue at here. After custom the Accessible Name, Role, Description for RichTextBox, these properties doesn't change in inspect, whether we need to file a new issue for it?

After custom 
AccessibleRole =List
AccessibleName=richTextBoxTest
AccessibleDescription=This is a richTextBox

image

Comment (vladimir-krestov) [reply src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs:1065]: I think, yes. To test this case too and be sure that the bug is here at the moment, and test it when we implement UIA for `RichTextBox`.
Like as for `TreeView`, that you have added here

Comment (Amy-Li03) [reply src/System.Windows.Forms/tests/AccessibilityTests/CusAccNameTesting.Designer.cs:1065]: OK, I will add RichTextBox to test app.

**Review by Tanya-Solyanik — APPROVED** [review]

**Review by RussKie — APPROVED** [review]
Thank you

**Pull request merged**